packages/grafana-runtime/src/services/backendSrv.ts

```diff
@@ -39,6 +39,13 @@ export interface BackendSrv {
 
   // DataSource requests add hooks into the query inspector
   datasourceRequest(options: BackendSrvRequest): Promise<any>;
+
+  // Can be used to augment the request headers just like the backend service
+  // would anywhere else in the code base. (Obviously, there should be good
+  // reasons not to use the request API that this service provides.)
+  augmentRequestHeaders(headers: BackendSrvRequest['headers']): any;
+
+  augmentDataSourceRequestHeaders(headers: BackendSrvRequest['headers']): any;
```

Comment:
Not sure this belongs to the public api. Internally we can use the "real" backendSrv like we do here https://github.com/grafana/grafana/blob/02140fff2d15004e7310fa01663610797a3b90b8/public/app/features/search/components/MoveToFolderModal.tsx#L8 and then those methods are accessible.

Reply:
Fair enough, that's what I've been discussing with @aocenas and @davkal already and is why I'd like to move the worker into the backend service. :)